server/src/main/java/com/google/coffeehouse/servlets/UpdateClubServlet.java

```diff
@@ -0,0 +1,225 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to update a {@link Club} from Http POST Request Body (in JSON format),
+ * save it in database, and return it in JSON format.
+ */
+@WebServlet("/api/update-club")
+public class UpdateClubServlet extends HttpServlet {
+  /** The fields of the Club object that can be updated. */
+  public static final List<String> updateableClubFields = Arrays.asList(
+      Club.DESCRIPTION_FIELD_NAME,
+      Club.CONTENT_WARNINGS_FIELD_NAME);
+  /** The fields of the Book object that can be updated. */
+  public static final List<String> updateableBookFields = Arrays.asList(
+      Book.AUTHOR_FIELD_NAME,
+      Book.ISBN_FIELD_NAME,
+      Book.TITLE_FIELD_NAME);
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request cannot be parsed for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** Message to be logged when the body of the POST request cannot be parsed. */
+  public static final String LOG_BODY_ERROR_MESSAGE =
+      "Body unable to be parsed in UpdateClubServlet: ";
+  /** Message to be logged when the body of the POST request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in UpdateClubServlet: ";
+  /** Message to be logged when the body of the POST request does not have required fields. */
+  public static final String LOG_SECURITY_MESSAGE =
+      "Forbidden action attempted: ";
+  /** 
+   * The error string sent by the response object in doPost when the body of the
+   * POST request does not have a required field.
+   */
+  public static final String NO_FIELD_ERROR = "No \"%s\" found in JSON.";
+  /** 
+   * The error string sent by the response object in doPost when the user attempting
+   * to update the club does not have permissions to do so.
+   */
+  public static final String LACK_OF_PRIVILEGE_ERROR = "Person does not have required privileges.";
+
+  /** The logged error string when an an ID token fails verification. */
+  public static final String INVALID_ID_TOKEN_ERROR = 
+      "ID token failed verification or didn't exist: ";
+  /** Name of the key in the input JSON that corresponds to the Club object. */
+  public static final String CLUB_FIELD_NAME = "club";
+  /** Name of the key in the input JSON that corresponds to the update mask. */
+  public static final String UPDATE_MASK_FIELD_NAME = "updateMask";
+  /** Name of the key in the input JSON that corresponds to the ID token. */
+  public static final String ID_TOKEN_FIELD_NAME = "idToken";
+
+  private static final Gson gson = new Gson();
+  private static final HttpTransport transport = new NetHttpTransport();
+  private static final GsonFactory jsonFactory = GsonFactory.getDefaultInstance();
+  private GoogleIdTokenVerifier verifier =
```

Comment:
Consider moving initialization of the verifier into the default constructor, similar to storageHandler.

Reply:
Good point, this has been moved :).